File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def zeroCnt(s):
    return s.count("0")

def filterZero(s):
    return [i for i in s if i != "0"]

def toBin(s):
    return bin(s)[2:]
    
def toInt(s):
    return int(s, 2)

def solution(s):
    
    cnt = 0
    zero = 0
    
    while True:
        cnt += 1
        zero += zeroCnt(s)
        s = filterZero(s)
        s = toBin(len(s))
        
        if s == "1":
            break
    return [cnt, zero]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let cnt = 0;
    let zero = 0;
    while (s !== "1"){
        let nonZero = s.split("").filter(v => v !== "0")
        zero += s.length - nonZero.length;
        s = nonZero.length .toString(2)
        cnt++;
    }
    
    return [cnt, zero]
}